spec.bs

```diff
@@ -18,3 +18,50 @@ Default Biblio Status: current
 Markup Shorthands: markdown yes
 Assume Explicit For: yes
 </pre>
+
+
+<h2 id="advertisements">Advertisements</h2>
+
+An <dfn export>advertisement</dfn>
+is a [=struct=] with the following [=struct/items=]:
+
+* An <dfn for=advertisement>render URL</dfn>. The render URL must have “https” as the scheme.
+* An optional <dfn for=advertisement>metadata</dfn> which is text containing opaque JSON encoded data, passed as an object to auction worklet.
+
+
+<h2 id="advertisement-interest-groups">Advertisement Interest Groups</h2>
+
+An <dfn export>advertisement interest group</dfn>
+is a [=struct=] with the following [=struct/items=]:
+ 
+* An <dfn for="advertisement interest group">expiration</dfn> date.   base::Time expiry;
+
+* An <dfn for="advertisement interest group">owner origin</dfn>, an [=origin=] url.
+
+* A <dfn for="advertisement interest group">name</dfn>, a [=string=].
+ 
+* A <dfn for="advertisement interest group">priority</dfn>, a double, initiallized to 0.
+ 
+* An <dfn for="advertisement interest group">enable_bidding_signals_prioritization</dfn>, a boolean, initialized to false.

```

Comment:
Should "boolean" be "[=boolean=]"?

Reply:
yes.  Can't do that for [=double=], however.  But xmp elements with class="idl" can use double, and it auto-links correctly.  Not sure why, or where the docs are.